src/execution/__tests__/defer-test.ts

```diff
@@ -472,6 +476,40 @@ describe('Execute: defer directive', () => {
     });
   });
 
+  it('Emits children of empty defer fragments', async () => {
+    const document = parse(`
+      query HeroNameQuery {
+        hero {
+          ... @defer {
+            ... @defer {
+              name
+            }
+          }
+        }
+      }
+    `);
+    const result = await complete(document);
+    expectJSON(result).toDeepEqual([
+      {
+        data: {
+          hero: {},
+        },
+        pending: [{ id: '0', path: ['hero'] }],
+        hasNext: true,
+      },
+      {
+        pending: [{ id: '1', path: ['hero'] }],
+        completed: [{ id: '0' }],
+        hasNext: true,
```

Comment:
this payload is unexpected to me. Since the outer @defer has no fields, shouldn't it be skipped?

Reply:
Do we need to merge https://github.com/graphql/graphql-js/pull/3993 first?

Reply:
Done => and rebased.